packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergEmbedWebViewActivity.java

```diff
@@ -80,7 +77,7 @@ public void onProgressChanged(WebView view, int progress) {
 
     protected void load() {
         String content = getIntent().getExtras().getString(ARG_CONTENT);
-        mWebView.loadData(content, "text/html", "UTF-8");
+        mWebView.loadDataWithBaseURL(null, content, "text/html", "UTF-8", null);
```

Comment:
The [documentation of `loadData`](https://developer.android.com/reference/android/webkit/WebView#loadData(java.lang.String,%20java.lang.String,%20java.lang.String)) mentions that `loadDataWithBaseURL` should be used when you need to identify the main frame's origin in a trustworthy way. I haven't found references to `loadDataWithBaseURL` relate to the encoding, so I'm wondering @jhnstn if you could share more info about the encoding and this function to have a better context.

Reply:
Sure thing @fluiddot 
Real quick, the root issue is the encoding. Specifically I found that  `#` symbols properly encoded as `%23` is the `content` resulted in `#` in the `WebView` .
( There is a note about `#` characters in the `loadData` [docs](https://developer.android.com/reference/android/webkit/WebView#loadData(java.lang.String,%20java.lang.String,%20java.lang.String)) but that was not the issue I was seeing. )

I found quite a few references to folks having to use `loadDataWithBaseUrl` to fix encoding issues. I've tried all the options I could find to get `loadData` to encode properly but nothing seemed to work.  This seemed to follow many of the [observations from this SO post](https://stackoverflow.com/questions/7412763/string-encoding-problem-in-webview). There are a few workarounds but seems like `loadDataWithBaseUrl` is the most straight forward and stable solution, here is a good [example](https://stackoverflow.com/a/4249537/1373043).

Then there is this [issue on Google](https://issuetracker.google.com/issues/36906575#c23) which has the status _won't  fix_ . There too folks recommend using `loadDataWithBaseUrl` as the best option. Someone also mentions that documentation of `loadData` is misleading.  

So I can't explain why `loadData` is not working based on the documentation and I haven't looked into the source code to see what's the difference between the two functions regarding encoding. But from what I've read from others with the same issue and by what I've tried with `loadData` I'm convinced that this is an appropriate fix. If this is not sufficient then I'm happy to find another solution. That said this is working

Reply:
Thanks @jhnstn for elaborating on the solution, I really appreciate it 🙇. It's interesting how `loadData` and `loadDataWithBaseUrl` functions behave so differently in terms of encoding, probably as you pointed out, there's something in the source code that processes the encoding in a different way (probably better 😅).

Reply:
I debugged the WebView using each function and confirmed the issue:
- `iframe` source URL with `loadData` ❌ :
```
https://video.wordpress.com/embed/<VIDEOPRESS_GUID>?cover=1&loop=1&preloadContent=metadata&sbc=
```

**Note how the URL stops at the seek bar color query param (`sbc`).**

- `iframe` source URL with `loadDataWithBaseUrl` ✅ :
```
https://video.wordpress.com/embed/<VIDEOPRESS_GUID>?cover=1&loop=1&preloadContent=metadata&sbc=%23f78da7&hd=1&metadata_token=<TOKEN>
```